mkosi

```diff
@@ -982,6 +983,70 @@ def invoke_dnf(args, workspace, repositories, base_packages, boot_packages, conf
     with mount_api_vfs(args, workspace):
         run(cmdline, check=True)
 
+@complete_step('Installing Clear Linux')
+def install_clear(args, workspace, run_build_script):
+    if args.release == "latest":
+        release = "clear"
+    else:
+        release = "clear/"+args.release
+
+    root = os.path.join(workspace, "root")
+
+    packages = ['os-core'] + args.packages
+    if run_build_script:
+        packages.extend(args.build_packages)
+    if args.bootable:
+        packages += ['kernel-native']
+
+    # Since swupd-extract program is usually installed with "go get",
+    # try to find it in some other locations besides PATH.
+    swupd_extract = None
+    sudo_user = os.getenv("SUDO_USER")
+    if sudo_user:
+        sudo_user_program = os.path.expanduser("~{}/go/bin/swupd-extract".format(sudo_user))
+        if os.path.exists(sudo_user_program):
+            swupd_extract = sudo_user_program
+    if swupd_extract is None:
+        user_program = os.path.expanduser("~/go/bin/swupd-extract")
```

Comment:
is that path in any way standardized? can't clearlinux just install this stuff like a normal binary like everybody else?

Reply:
The problem is actually installing on other distros so I can generate a Clear image from my Fedora host. The path is standard for go programs.